### PR TITLE
Hackathon-ready

### PR DESCRIPTION
**Pull request by olegoaer**
Preparation of the documentation to drive the contributors to the Android plugin

### CODE REVIEW

File: android-plugin/RULES.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Android-specific rules rely on a multi-scope scanning, including Java source files, Xml files, Gradle files, and more generally the project structure.
+
+### 🍃 Environment
+
+The set of rules comes from the detailed [Energy Smells catalog](https://olegoaer.perso.univ-pau.fr/android-energy-smells/). 3/4 of them have been already implemented in the plugin. Table of unimplemented rules below:
+
+| # | **Rule Name**      |     **Scanner**     |      **Observation**     |
+|---|:----------------|:-------------:|:-------------:|
+| ELEA001 | Everlasting Service        | Java | Requires `PostProjectAnalysisTask()` callback |
```
Comment (jhertout) [inline android-plugin/RULES.md:11]: I think that we should explain somewhere what "Requires `PostProjectAnalysisTask()` callback" means.

File: android-plugin/RULES.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+| EBOT004 | Uncached Data Reception       | Java | Requires `PostProjectAnalysisTask()` callback |
+| ESOB003/ESOB004 | Dark UI      | Java, Xml | Partially implemented (e.g. [1](./src/main/java/io/ecocode/xml/checks/sobriety/DarkUIBrightColorsXmlRule.java), [2](./src/main/java/io/ecocode/xml/checks/sobriety/DarkUIThemeXmlRule.java)) |
+| ESOB009 | Day Night Mode     | File System, Xml | Requires `PostProjectAnalysisTask()` callback |
+| ESOB015 | Animation-free | Java, Xml, File System |  |
+| EPOW008 | Constrainted Worker | Java |  |
+| EBAT001 | Service@Boot-time    | Java, Xml  | Likely detectable in Xml only |
+| EREL004 | Same dependencies    | Gradle | Requires a knowledge base |
```
Comment (jhertout) [inline android-plugin/RULES.md:18]: same thing as before: what "Requires a knowledge base" means?

File: android-plugin/RULES.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+### 🍃 Environment
+
+The set of rules comes from the detailed [Energy Smells catalog](https://olegoaer.perso.univ-pau.fr/android-energy-smells/). 3/4 of them have been already implemented in the plugin. Table of unimplemented rules below:
+
+| # | **Rule Name**      |     **Scanner**     |      **Observation**     |
+|---|:----------------|:-------------:|:-------------:|
+| ELEA001 | Everlasting Service        | Java | Requires `PostProjectAnalysisTask()` callback |
+| EBOT004 | Uncached Data Reception       | Java | Requires `PostProjectAnalysisTask()` callback |
+| ESOB003/ESOB004 | Dark UI      | Java, Xml | Partially implemented (e.g. [1](./src/main/java/io/ecocode/xml/checks/sobriety/DarkUIBrightColorsXmlRule.java), [2](./src/main/java/io/ecocode/xml/checks/sobriety/DarkUIThemeXmlRule.java)) |
```
Comment (jhertout) [inline android-plugin/RULES.md:13]: For this kind of partialy implemented rule, we should give more precision (what is done and what we expect)

File: android-plugin/RULES.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+| ESOB009 | Day Night Mode     | File System, Xml | Requires `PostProjectAnalysisTask()` callback |
+| ESOB015 | Animation-free | Java, Xml, File System |  |
+| EPOW008 | Constrainted Worker | Java |  |
+| EBAT001 | Service@Boot-time    | Java, Xml  | Likely detectable in Xml only |
+| EREL004 | Same dependencies    | Gradle | Requires a knowledge base |
+| EREL005 | Duplicate dependencies    | Gradle | Requires a knowledge base |
+| EREL007 | Clear cache    | Java | Method `deleteRecursively()` is Kotlin-only |
```
Comment (jhertout) [inline android-plugin/RULES.md:20]: Say that it can't be done in Java?

File: android-plugin/RULES.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+| EREL008 | Convert to WebP | File System |  |
+| EREL009 | Shrink Resources    | Gradle |  |
+| EREL010 | Disable Obfuscation    | Gradle |  |
+
+
+### 🤝 Social
+
+The set of rules comes from the detailed [Social Smells catalog](https://olegoaer.perso.univ-pau.fr/android-social-smells/index.html) (work in progress). 0 rules have been implemented so far in the plugin. Table of unimplemented rules below:
```
Comment (jhertout) [inline android-plugin/RULES.md:28]: 0 rules have bee -> 0 rule has been?

File: android-plugin/RULES.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+| # | **Rule Name**      |     **Scanner**     |      **Observation**     |
+| ---|:----------------|:-------------:|:-------------:|
+| SPRI001 | Crashlytics automatic opt-in       | Java, Xml |  |
+| SPRI002 | Google Tracker | Java | Partially implemented. See [PR#1](https://github.com/green-code-initiative/ecoCode-mobile/pull/1) |
```
Comment (jhertout) [inline android-plugin/RULES.md:33]: For this kind of partialy implemented rule, we should give more precision (what is done and what we expect)

File: android-plugin/RULES.md (hunk 3 above)
Comment (olegoaer) [reply android-plugin/RULES.md:13]: I don't remember why this rule wasn't completed. Any idea @Jberque?

File: android-plugin/RULES.md (hunk 4 above)
Comment (olegoaer) [reply android-plugin/RULES.md:20]: As I understand it, this method exists in Java in Kotlin but not in Java 🤷
I don't know if it exists a workaround for that issue.

File: android-plugin/RULES.md (hunk 6 above)
Comment (olegoaer) [reply android-plugin/RULES.md:33]: This rule seems finished but the PR must be reworked by Snapp'. Hence, I guess this rule is intended to be deleted from the table.

**Review by dedece35 — DISMISSED** [review]

File: android-plugin/RULES.md (hunk 1 above)
Comment (olegoaer) [reply android-plugin/RULES.md:11]: It seems this method appeared in Sonar API 8.9. If so, that's good news, isn't it @Jberque?

**Pull request merged**